src/CodeStyle/CSharp/CodeFixes/EditorConfigOptionsApplier.cs

```diff
@@ -0,0 +1,92 @@
+            var csharpOptionsType = typeof(CSharp.Formatting.CSharpFormattingOptions);
+            _formattingOptionsWithStorage = GetOptionsWithStorageFromTypes(new[] { commonOptionsType, csharpOptionsType });
+        }
+
+        public OptionSet ApplyConventions(OptionSet optionSet, ICodingConventionsSnapshot codingConventions, string languageName)
+        {
+            foreach (var optionWithStorage in _formattingOptionsWithStorage)
+            {
+                if (TryGetConventionValue(optionWithStorage, codingConventions, out var value))
+                {
+                    var option = optionWithStorage.Item1;
+                    var optionKey = new OptionKey(option, option.IsPerLanguage ? languageName : null);
+                    optionSet = optionSet.WithChangedOption(optionKey, value);
+                }
+            }
+
+            return optionSet;
+        }
+
+        private OptionSet ApplyConventionsForOptions(OptionSet optionSet, IEnumerable<(IOption, OptionStorageLocation, MethodInfo)> optionsWithStorage, ICodingConventionsSnapshot codingConventions, string languageName)
+        {
+            foreach (var optionWithStorage in optionsWithStorage)
+            {
+                if (TryGetConventionValue(optionWithStorage, codingConventions, out var value))
+                {
+                    var option = optionWithStorage.Item1;
+                    var optionKey = new OptionKey(option, option.IsPerLanguage ? languageName : null);
+                    optionSet = optionSet.WithChangedOption(optionKey, value);
+                }
+            }
+
+            return optionSet;
+        }
+
+        internal IReadOnlyList<(IOption, OptionStorageLocation, MethodInfo)> GetOptionsWithStorageFromTypes(params Type[] formattingOptionTypes)
+        {
+            var optionType = typeof(IOption);
+            return formattingOptionTypes
+                .SelectMany(t => t.GetProperties(BindingFlags.Public | BindingFlags.Static | BindingFlags.GetProperty))
+                .Where(p => optionType.IsAssignableFrom(p.PropertyType))
+                .Select(p => (IOption)p.GetValue(null))
+                .Select(GetOptionWithStorage)
+                .Where(ows => ows.Item2 != null)
+                .ToList();
+        }
+
+        internal (IOption, OptionStorageLocation, MethodInfo) GetOptionWithStorage(IOption option)
+        {
+            var editorConfigStorage = !option.StorageLocations.IsDefaultOrEmpty
+                ? option.StorageLocations.FirstOrDefault(IsEditorConfigStorage)
+                : null;
+            var tryGetOptionMethod = editorConfigStorage?.GetType().GetMethod("TryGetOption");
+            return (option, editorConfigStorage, tryGetOptionMethod);
+        }
+
+        internal static bool IsEditorConfigStorage(OptionStorageLocation storageLocation)
+        {
+            return storageLocation.GetType().FullName.StartsWith("Microsoft.CodeAnalysis.Options.EditorConfigStorageLocation");
```

Comment:
It is not.

Reply:
> if we need something, then it's important enough to make it a public API.)

But that's not what's going on here :)  

This reflection stuff worries me.  it's effectively no better/worse than IVT.  it's basically getting privileged access, just through another means.

Reply:
> But that's not what's going on here

It's not part of this pull request, but it's a fundamental requirement of eliminating the Workspace layer dependency during command line compilation so it will go away.

Reply:
➡️ Marking this resolved. The dependency on 15.7 assemblies removes the viability of the IVT approach so the current one is left.